src/tagged-type.ts

```diff
@@ -253,9 +284,109 @@ const encode = (input: QueryValue): QueryValue => {
         return encodeMap["set"](input);
       } else if (input instanceof EmbeddedSet) {
         return encodeMap["set"](input);
+      } else if (input instanceof Query) {
+        throw new TypeError(
+          "Cannot encode instance of type 'Query'. Try using TaggedTypeFormat.encodeInterpolation instead."
+        );
       } else {
         return encodeMap["object"](input);
       }
   }
   // anything here would be unreachable code
 };
+
+const encodeInterpolation = (input: QueryValue): QueryInterpolation => {
+  if (input === undefined) {
+    throw new TypeError("Passing undefined as a QueryValue is not supported");
+  }
+  switch (typeof input) {
+    case "bigint":
+      return encodeValueInterpolation(encodeMap["bigint"](input));
+    case "string":
+      return encodeValueInterpolation(encodeMap["string"](input));
+    case "number":
+      return encodeValueInterpolation(encodeMap["number"](input));
+    case "boolean":
+      return encodeValueInterpolation(input);
+    case "object":
+      if (input == null) {
+        return encodeValueInterpolation(null);
+      } else if (input instanceof Date) {
+        return encodeValueInterpolation(encodeMap["date"](input));
+      } else if (input instanceof DateStub) {
+        return encodeValueInterpolation(encodeMap["faunadate"](input));
+      } else if (input instanceof TimeStub) {
+        return encodeValueInterpolation(encodeMap["faunatime"](input));
+      } else if (input instanceof Module) {
+        return encodeValueInterpolation(encodeMap["module"](input));
+      } else if (input instanceof Document) {
+        // Document extends DocumentReference, so order is important here
+        return encodeValueInterpolation(encodeMap["document"](input));
+      } else if (input instanceof DocumentReference) {
+        return encodeValueInterpolation(encodeMap["documentReference"](input));
+      } else if (input instanceof NamedDocument) {
+        // NamedDocument extends NamedDocumentReference, so order is important here
+        return encodeValueInterpolation(encodeMap["namedDocument"](input));
+      } else if (input instanceof NamedDocumentReference) {
+        return encodeValueInterpolation(
+          encodeMap["namedDocumentReference"](input)
+        );
+      } else if (input instanceof NullDocument) {
+        return encodeInterpolation(input.ref);
+      } else if (input instanceof Page) {
+        return encodeValueInterpolation(encodeMap["set"](input));
+      } else if (input instanceof EmbeddedSet) {
+        return encodeValueInterpolation(encodeMap["set"](input));
```

Comment:
same for all of this. it seems like theres a lot duplicated between this and `encode`.